arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -3244,20 +3246,51 @@ ErrorCode RestReplicationHandler::createCollection(VPackSlice slice) {
     }
   }
   patch.add(StaticStrings::ObjectId, VPackSlice::nullSlice());
-  patch.add("cid", VPackSlice::nullSlice());
+  patch.add(StaticStrings::DataSourceCid, VPackSlice::nullSlice());
   patch.add(StaticStrings::DataSourceId, VPackSlice::nullSlice());
+  if (ServerState::instance()->isSingleServer()) {
+    // needed in case we restore cluster related data into single server instance
+    patch.add(StaticStrings::DataSourcePlanId, VPackSlice::nullSlice());
+  }
   patch.close();
 
   VPackBuilder builder =
       VPackCollection::merge(slice, patch.slice(),
                              /*mergeValues*/ true, /*nullMeansRemove*/ true);
   slice = builder.slice();
 
-  col = _vocbase.createCollection(slice);
+  // Initializing creation options
+  TRI_col_type_e collectionType =
+      Helper::getNumericValue<TRI_col_type_e, int>(slice, StaticStrings::DataSourceType,
+                                                   TRI_COL_TYPE_UNKNOWN);
+  std::vector<CollectionCreationInfo> infos{{name, collectionType, slice}};
+  bool isNewDatabase = false;
+  bool allowSystem = true;
+  bool isSingleServerEnterpriseCollection = false;
+  bool enforceReplicationFactor = false;
 
-  if (col == nullptr) {
+#ifdef USE_ENTERPRISE
+  if ((slice.hasKey(StaticStrings::IsSmart) && slice.get(StaticStrings::IsSmart).isBoolean() &&
+       slice.get(StaticStrings::IsSmart).isTrue()) ||
```

Comment:
Just a suggestion: Could we add helper methods into `VelocyPack` slice objects like the following:
```
slice.isTruthy(StaticStrings::IsSmart)
```

This will work identical to the 3-boolean conjunction define above in this case, but will also generally return true for any truthy value found for the provided key.

Reply:
![image](https://user-images.githubusercontent.com/1244163/140733562-6be911ca-0bc5-48e6-a721-dfaae7c4c5a0.png)

Good catch. This can be used as this. Fixed.